plugin/storage/cassandra/spanstore/writer_test.go

```diff
@@ -352,23 +353,36 @@ func TestStorageMode_IndexOnly_WithFilter(t *testing.T) {
 
 func TestStorageMode_IndexOnly_FirehoseSpan(t *testing.T) {
 	withSpanWriter(0, func(w *spanWriterTest) {
-
-		w.writer.serviceNamesWriter = func(serviceName string) error { return nil }
-		w.writer.operationNamesWriter = func(operation dbmodel.Operation) error { return nil }
+		serviceWritten := atomic.NewString("")
+		operationWritten := &atomic.Value{}
+		w.writer.serviceNamesWriter = func(serviceName string) error {
+			serviceWritten.Store(serviceName)
+			return nil
+		}
+		w.writer.operationNamesWriter = func(operation dbmodel.Operation) error {
+			operationWritten.Store(operation)
+			return nil
+		}
 		span := &model.Span{
-			TraceID: model.NewTraceID(0, 1),
+			TraceID:       model.NewTraceID(0, 1),
+			OperationName: "package-delivery",
 			Process: &model.Process{
-				ServiceName: "service-a",
+				ServiceName: "planet-express",
 			},
 			Flags: model.Flags(8),
 		}
 
 		err := w.writer.WriteSpan(span)
 		assert.NoError(t, err)
 		w.session.AssertExpectations(t)
-		w.session.AssertNotCalled(t, "Query", stringMatcher(serviceOperationIndex))
 		w.session.AssertNotCalled(t, "Query", stringMatcher(serviceNameIndex))
 		w.session.AssertNotCalled(t, "Query", stringMatcher(durationIndex))
+		assert.Equal(t, "planet-express", serviceWritten.Load())
+		assert.Equal(t, dbmodel.Operation{
+			ServiceName:   "planet-express",
+			SpanKind:      "",
+			OperationName: "package-delivery",
+		}, operationWritten.Load())
```

Comment:
Are these not tested elsewhere? For e.g., if plugin/storage/cassandra/spanstore/writer.go L183 were removed - wouldn't other tests fail?

Reply:
I am not testing whether index writing works, but whether it's invoked for firehose span. This is the only test that makes distinction between two types of indices.